### ISSUE TITLE
numpy.testing.assert_arrays_equal omits NaNs when printing output

### ISSUE THREAD

**Issue by tigra** [numpy/numpy#12874]
When asserting array equality using `assert_arrays_equal`, and arrays aren't actually equal, the verbose output showing unequal arrays is printed. If one of or both arrays contain NaNs, the corresponding location is omitted from both arrays.
Mismatch percentage is also calculated incorrectly, ignoring NaNs.

Actual output:
```
(mismatch 100.0%)
 x: array([1.])
 y: array([2.])
```
Expected output:
```
(mismatch 50.0%)
 x: array([1., nan])
 y: array([2., nan])
```

### Reproducing code example:

```python
import numpy as np
from numpy.testing import assert_array_equal
assert_array_equal(np.array([1, np.nan]), np.array([2, np.nan]))
```

### Error message:
```
Traceback (most recent call last):
  File "/usr/lib/python3.6/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib/python3.6/unittest/case.py", line 605, in run
    testMethod()
  File "/home/tigra/Ares/venv/lib/python3.6/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/home/tigra/Dropbox/Ares/ks_tmp/tests/problem_test.py", line 87, in test_numpy
    assert_array_equal(np.array([1, np.nan]), np.array([2, np.nan]))
  File "/home/tigra/Ares/venv/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 865, in assert_array_equal
    verbose=verbose, header='Arrays are not equal')
  File "/home/tigra/Ares/venv/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 789, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Arrays are not equal

(mismatch 100.0%)
 x: array([1.])
 y: array([2.])
```
### Numpy/Python version information:
1.15.4 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0]

**Comment by AnnaMelk** [comment]
The issue is resolved in the Numpy 1.16.2 version. It both shows Nan in the arrays and has correct mismatch percentage.

**Comment by mattip** [comment]
Closing thanks @AnnaMelk
